infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/transport/sample/CorfuStreamObserver.java

```diff
@@ -0,0 +1,68 @@
+package org.corfudb.infrastructure.logreplication.transport.sample;
+
+import io.grpc.Status;
+import io.grpc.StatusException;
+import io.grpc.StatusRuntimeException;
+import io.grpc.stub.StreamObserver;
+
+public class CorfuStreamObserver<T> implements StreamObserver<T> {
+
+    private final StreamObserver<T> observer;
+
+    public CorfuStreamObserver(StreamObserver<T> observer) {
+        this.observer = observer;
+    }
+
+    /**
+     * Receives a value from the stream.
+     *
+     * <p>Can be called many times but is never called after {@link #onError(Throwable)} or {@link
+     * #onCompleted()} are called.
```

Comment:
Previously I selected “inherit java docs” so those comments were actually for `StreamObserver`. Maybe we need a class level doc stating that this class is for addressing the current issue with native `StreamObserver`?

Reply:
Added a class level comment, and kept only the first line of the method level comment.